mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -121,48 +120,57 @@ pub struct TendermintConf {
 
 impl TendermintConf {
     pub fn try_from_json(ticker: &str, conf: &Json) -> MmResult<Self, TendermintInitError> {
-        let avg_block_time = conf["avg_block_time"].as_i64().unwrap_or(0);
+        let avg_blocktime = conf.get("avg_blocktime").ok_or({
+            TendermintInitError {
+                ticker: ticker.to_string(),
+                kind: TendermintInitErrorKind::AvgBlockTimeMissing,
+            }
+        })?;
 
-        // `avg_block_time` can not be less than 1 OR bigger than 255(u8::MAX)
-        if avg_block_time < 1 || avg_block_time > std::u8::MAX as i64 {
-            return MmError::err(TendermintInitError {
+        let avg_blocktime = avg_blocktime.as_i64().ok_or({
+            TendermintInitError {
                 ticker: ticker.to_string(),
-                kind: TendermintInitErrorKind::AvgBlockTimeMissingOrInvalid,
-            });
-        }
+                kind: TendermintInitErrorKind::AvgBlockTimeInvalid,
+            }
+        })?;
+
+        let avg_blocktime = u8::try_from(avg_blocktime).map_err(|_| TendermintInitError {
+            ticker: ticker.to_string(),
+            kind: TendermintInitErrorKind::AvgBlockTimeInvalid,
+        })?;
 
         let derivation_path = json::from_value(conf["derivation_path"].clone()).map_to_mm(|e| TendermintInitError {
             ticker: ticker.to_string(),
             kind: TendermintInitErrorKind::ErrorDeserializingDerivationPath(e.to_string()),
         })?;
 
         Ok(TendermintConf {
-            avg_block_time: avg_block_time as u8,
+            avg_blocktime,
             derivation_path,
         })
     }
 }
 
 pub struct TendermintCoinImpl {
     ticker: String,
-    /// TODO
-    /// Test Vec<String(rpc_urls)> instead of HttpClient and pick
-    /// better one in terms of performance & resource consumption on runtime.
     rpc_clients: Vec<HttpClient>,
+    /// As seconds
+    avg_blocktime: u8,
     /// My address
-    avg_block_time: u8,
     pub account_id: AccountId,
     pub(super) account_prefix: String,
     priv_key: Vec<u8>,
-    decimals: u8,
+    pub(crate) decimals: u8,
     pub(super) denom: Denom,
     chain_id: ChainId,
     gas_price: Option<f64>,
     pub(super) sequence_lock: AsyncMutex<()>,
-    tokens_info: PaMutex<HashMap<String, ActivatedTokenInfo>>,
+    pub(crate) tokens_info: PaMutex<HashMap<String, ActivatedTokenInfo>>,
     /// This spawner is used to spawn coin's related futures that should be aborted on coin deactivation
     /// or on [`MmArc::stop`].
     pub(super) abortable_system: AbortableQueue,
+    pub(crate) history_sync_state: Mutex<HistorySyncState>,
+    pub ctx: MmWeak,
```

Comment:
Is it possible somehow to avoid storing `MmWeak` at the `TendermintCoin` structure? Sometime ago we had a TODO comment in eth.rs to get rid of `EthCoin::ctx` as well, but can't find it...